catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/KmzTransformerImpl.java

```diff
@@ -0,0 +1,129 @@
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KmzTransformerImpl.class);
+
+  static {
+    try {
+      KMZ_MIMETYPE = new MimeType("application/vnd.google-earth.kmz");
+    } catch (MimeTypeParseException e) {
+      throw new ExceptionInInitializerError(e);
+    }
+  }
+
+  private KmlTransformer kmlTransformer;
+
+  public KmzTransformerImpl(KmlTransformer kmlTransformer) {
+    this.kmlTransformer = kmlTransformer;
+  }
+
+  /**
+   * Transforms a {@link SourceResponse} to a zipped KMZ file.
+   *
+   * @param upstreamResponse
+   * @param arguments the arguments that may be used to execute the transform
+   * @return
+   * @throws CatalogTransformerException
+   */
+  @Override
+  public BinaryContent transform(
+      SourceResponse upstreamResponse, Map<String, Serializable> arguments)
+      throws CatalogTransformerException {
+    BinaryContent unzippedKml = kmlTransformer.transform(upstreamResponse, arguments);
+    BinaryContent kmz = kmlToKmzTransform(unzippedKml);
+
+    if (kmz == null) {
+      throw new CatalogTransformerException(String.format("Unable to transform KML to KMZ."));
+    }
+    return kmz;
+  }
+
+  /**
+   * Transforms a single metacard to a zipped KMZ file.
+   *
+   * @param metacard the {@link Metacard} to be transformed
+   * @param arguments any arguments to be used in the transformation. Keys are specific to each
+   *     {@link MetacardTransformer} implementation
+   * @return
+   * @throws CatalogTransformerException
+   */
+  @Override
+  public BinaryContent transform(Metacard metacard, Map<String, Serializable> arguments)
+      throws CatalogTransformerException {
+    BinaryContent unzippedKml = kmlTransformer.transform(metacard, arguments);
+    BinaryContent kmz = kmlToKmzTransform(unzippedKml);
+
+    if (kmz == null) {
+      throw new CatalogTransformerException(
+          String.format("Unable to transform to KMZ for metacard ID: %s", metacard.getId()));
+    }
+    return kmz;
+  }
+
+  /**
+   * Converts an unzipped KML file packaged as a {@link BinaryContent} to a zipped KMZ file.
+   *
+   * @param unzippedKml - unzipped kml {@link BinaryContent}
+   * @return BinaryContent - zipped KML file containing KML data.
+   */
+  public BinaryContent kmlToKmzTransform(BinaryContent unzippedKml) {
+    try {
+      InputStream inputStream = unzippedKml.getInputStream();
+      TemporaryFileBackedOutputStream temporaryFileBackedOutputStream =
+          new TemporaryFileBackedOutputStream();
+      ZipOutputStream zipOutputStream = new ZipOutputStream(temporaryFileBackedOutputStream);
+
+      final ZipEntry e = new ZipEntry(DOC_KML);
+      zipOutputStream.putNextEntry(e);
+      IOUtils.copy(inputStream, zipOutputStream);
+      zipOutputStream.closeEntry();
+      zipOutputStream.finish();
+      final InputStream zipFile = temporaryFileBackedOutputStream.asByteSource().openStream();
+      final BinaryContentImpl binaryContent = new BinaryContentImpl(zipFile, KMZ_MIMETYPE);
```

Comment:
✏️ You could return right here since we don't modify `binaryContent` after this line.